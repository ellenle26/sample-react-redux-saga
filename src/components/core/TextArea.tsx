import classNames from "classnames";
import { memo } from "react"

const TextArea: React.FC<TextAreaProps> = (props) => {
    const {title, value, fullWidth, rows, onChange, className} = props;
    return <div className={classNames(className)}>
        {title && <p className="mb-2 text-base">{title}</p>}
        <textarea rows={rows ? rows: 20}
        onChange = {e => onChange && onChange(e.target.value)} 
        className={classNames("outline-none border-solid border", {"w-full": fullWidth === undefined || fullWidth === true})} 
        value={value}/>
    </div>
}

interface TextAreaProps {
    className?: string;
    title?: string;
    value?: string;
    fullWidth?: boolean;
    rows?: number;
    onChange?: (value: string) => void;
}

export default memo(TextArea);

