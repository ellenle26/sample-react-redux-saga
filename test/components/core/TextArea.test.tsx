/**
 * @jest-environment jsdom
*/

import TextArea from "@components/core/TextArea";
import {fireEvent, render} from "@testing-library/react";
import React from "react";

test("snapshot testing", () => {
  const renderResult = render(<TextArea />);
  expect(renderResult.asFragment()).toMatchSnapshot();
});

test("title testing", () => {
    const renderResult = render(<TextArea title="Title" />);
    const pTags = renderResult.container.getElementsByTagName("p");
    expect(pTags.length).toEqual(1);
    expect(pTags[0].innerHTML).toMatch("Title");
});

test("full width testing", () => {
    const renderResult = render(<TextArea title="Title" fullWidth = {true} />);
    const textAreaTags = renderResult.container.getElementsByTagName("textarea");
    expect(textAreaTags[0].className).toContain("w-full");
});

test("rows testing", () => {
    const renderResult = render(<TextArea title="Title" rows={50} />);
    const textAreaTags = renderResult.container.getElementsByTagName("textarea");
    expect(textAreaTags[0].rows).toEqual(50);
});

test("change event testing", () => {
    const handleChange = jest.fn();
    const renderResult = render(<TextArea onChange={value => handleChange(value)} />);
    const textAreaTags = renderResult.container.getElementsByTagName("textarea");
    fireEvent.change(textAreaTags[0], {target: {value: "this is text"}});

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toBeCalledWith("this is text");
  });