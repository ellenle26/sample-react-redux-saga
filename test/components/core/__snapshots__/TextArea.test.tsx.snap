// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <textarea
      class="outline-none border-solid border w-full"
      rows="20"
    />
  </div>
</DocumentFragment>
`;
